web-common/src/features/dashboards/filters/measure-filters/MeasureFilter.svelte

```diff
@@ -1,34 +1,148 @@
 
   let active = !filter;
 
+  const initialValues = {
+    dimension: dimensionName,
+    operation: filter?.operation ?? MeasureFilterOperationOptions[0].value,
+    value1: filter?.value1 ?? "",
+    value2: filter?.value2 ?? "",
+  };
+
+  const validationSchema = object().shape({
+    dimension: string().required("Required"),
+    operation: mixed<MeasureFilterOperation>()
+      .oneOf(Object.values(MeasureFilterOperation))
+      .required("Required"),
+    value1: string()
+      .required("Required")
+      .test("is-numeric", "Value must be a valid number", (value) => {
+        return !isNaN(Number(value)) && value.trim() !== "";
+      }),
+    value2: string().when("operation", {
+      is: (val: MeasureFilterOperation) => expressionIsBetween(val),
+      then: (schema) =>
+        schema
+          .required("Required")
+          .test("is-numeric", "Value must be a valid number", (value) => {
+            return !isNaN(Number(value)) && value.trim() !== "";
+          }),
+      otherwise: (schema) => schema.optional(),
+    }),
+  });
+
+  const { form, errors, submit, enhance, reset } = superForm(
+    defaults(initialValues, yup(validationSchema)),
+    {
+      SPA: true,
+      validators: yup(validationSchema),
+      onUpdate({ form }) {
+        if (!form.valid) return;
+        const values = form.data;
+
+        onApply({
+          dimension: values.dimension,
+          oldDimension: dimensionName,
+          filter: {
+            measure: name,
+            operation: values.operation,
+            type: MeasureFilterType.Value,
+            value1: values.value1,
+            value2: values.value2 ?? "",
+          },
+        });
+
+        active = false;
+      },
+      invalidateAll: false,
+      resetForm: false,
+    },
+  );
+
+  $: ({ operation } = $form);
+
+  $: isBetweenExpression = expressionIsBetween(operation);
+
+  $: dimensionOptions =
+    allDimensions.map((d) => ({
+      value: d.name as string,
+      label: getDimensionDisplayName(d),
+    })) ?? [];
+
   function handleDismiss() {
     if (!filter) {
-      dispatch("remove");
+      onRemove();
     } else {
       active = false;
+      reset({
+        data: {
+          dimension: dimensionName,
+          operation:
+            filter?.operation ?? MeasureFilterOperationOptions[0].value,
+          value1: filter?.value1 ?? "",
+          value2: filter?.value2 ?? "",
+        },
+      });
     }
   }
+
+  function expressionIsBetween(op: MeasureFilterOperation | string) {
+    return (
+      isMFO(op) &&
+      (op === MeasureFilterOperation.Between ||
+        op === MeasureFilterOperation.NotBetween)
+    );
+  }
+
+  function isMFO(value: string): value is MeasureFilterOperation {
+    return value in MeasureFilterOperation;
+  }
 </script>
 
-<DropdownMenu.Root
+<svelte:window
+  on:keydown={(e) => {
+    if (e.key === "Escape") {
+      handleDismiss();
+    } else if (e.key === "Enter") {
+      submit();
```

Comment:
Why do we need this on top of `onEnter` for inputs? Would this interfere with other inputs outside of this form?

Reply:
We have @typescript-eslint/no-unsafe-enum-comparison set to warn. I haven't done the work to make our Input/Select components accept generic props (mostly because there isn't much benefit), so the change function where we need to use `expressionIsBetween` only knows that the new value is a string. Can't type cast in the Svelte template section yet and didn't want to pull the function into the script tag. So, I'm type guarding the string before making the enum comparison.